Makefile

```diff
@@ -275,6 +275,12 @@ endif
 CXX11_CHECK_CMD = $(CXX) -std=c++11 -o $(TMPOUT) -c test/build/c++11.cc
 HAS_CXX11 = $(shell $(CXX11_CHECK_CMD) 2> /dev/null && echo true || echo false)
 
+CHECK_SHADOW_WORKS_CMD = $(CC) -std=c99 -Werror -Wshadow -o $(TMPOUT) -c test/build/shadow.c
+HAS_WORKING_SHADOW = $(shell $(CHECK_SHADOW_WORKS_CMD) 2> /dev/null && echo true || echo false)
+ifeq ($(HAS_WORKING_SHADOW),true)
+W_SHADOW=-Wshadow
+endif
```

Comment:
Not sure this one works as expected. For example the following code below always returns true with or without the shadow warning parameter:

```Bash
$ `gcc -std=c99 -Werror -o /dev/null -c shadow.c 2> /dev/null` && echo true || echo false
true
$ `gcc -std=c99 -Werror -Wshadow -o /dev/null -c shadow.c 2> /dev/null` && echo true || echo false
true
```